core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -258,6 +260,15 @@ public void drawPlace(int x, int y, int rotation, boolean valid){
         Draw.reset();
     }
 
+    @Override
+    public void draw(Tile tile) {
+        super.draw(tile);
+        if(tile.entity.power.graph.getPowerNodeColor() == null || tile.block() instanceof PowerSource) return;
```

Comment:
`PowerSource` could override this method and draw the standard tile instead of having a special check here.

Reply:
I'll try to move it there.